shuffle-storage/src/main/java/com/alibaba/flink/shuffle/storage/partition/LocalFileMapPartitionFactory.java

```diff
@@ -223,13 +238,49 @@ public LocalFileMapPartitionMeta recoverDataPartitionMeta(DataInput dataInput)
     }
 
     @Override
-    public List<StorageMeta> getHddStorageMetas() {
-        return new ArrayList<>(hddStorageMetas);
+    public void updateUsableStorageSpace() {
+        if (ssdStorageMetas.isEmpty()) {
+            usableSpace.setSsdUsableSpaceBytes(0);
+        }
+        for (StorageMeta storageMeta : ssdStorageMetas) {
+            long usableSpaceBytes = storageMeta.updateUsableSpace();
+            if (usableSpaceBytes > usableSpace.getSsdUsableSpaceBytes()) {
+                usableSpace.setSsdUsableSpaceBytes(usableSpaceBytes);
```

Comment:
Maybe the logic here is not right. The `usableSpace` should be cleared before each update, otherwise, the value will not be updated and the old value will be retained. 
Other updated values have similar problems.